### PR TITLE
JAMES-3400 James Webadmin CLI domain delete command

### PR DESCRIPTION
**Pull request by chibenwa**
Reopens https://github.com/linagora/james-project/pull/3989

### CODE REVIEW

**Comment by chibenwa** [comment]
Please always use JAMES-3400 as a JIRA prefix for the CLI

**Review by chibenwa — COMMENTED** [review]
:+1:

**Comment by Arsnael** [comment]
```
[04a352ab858c588a5d334d587001e7cabd7cf5f2] [ERROR]   CassandraMailboxMapperTest$ConsistencyTest.createAfterPreviousDeleteOnFailedCreateShouldCreateAMailbox:604 Multiple Failures (1 failure)
[04a352ab858c588a5d334d587001e7cabd7cf5f2] 	
[04a352ab858c588a5d334d587001e7cabd7cf5f2] Expected size:<1> but was:<2> in:
[04a352ab858c588a5d334d587001e7cabd7cf5f2] <[Mailbox{id=cb08e930-1f47-11eb-929f-9be21bdc23ec, namespace=#private, user=Username{localPart=user, domainPart=Optional.empty}, name=INBOX},
[04a352ab858c588a5d334d587001e7cabd7cf5f2]     Mailbox{id=c8361a20-1f47-11eb-929f-9be21bdc23ec, namespace=#private, user=Username{localPart=user, domainPart=Optional.empty}, name=name}]>
[04a352ab858c588a5d334d587001e7cabd7cf5f2] at CassandraMailboxMapperTest$ConsistencyTest.lambda$createAfterPreviousDeleteOnFailedCreateShouldCreateAMailbox$34(CassandraMailboxMapperTest$ConsistencyTest.java:615)
```
test this please

**Review by Arsnael — APPROVED** [review]

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 1]
```diff
@@ -93,4 +93,32 @@ void domainCreateCommandWithInvalidNameShouldFailed(GuiceJamesServer server) {
...
+
+        WebAdminCli.executeFluent(new PrintStream(outputStreamCaptor), new PrintStream(errorStreamCaptor),
+            "--url", "http://127.0.0.1:" + port.getValue(), "domain", "list");
+
+        assertThat(exitCode).isEqualTo(0);
+        assertThat(outputStreamCaptor.toString().contains("linagora.com")).isFalse();
+    }
+
+    @Test
+    void domainDeleteCommandWithDefaultDomainShouldFailed(GuiceJamesServer server) {
```
Comment (Arsnael) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:115]: `domainDeleteCommandWithDefaultDomainShouldFail`

**Review by chibenwa — COMMENTED** [review]
:+1:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**